src/csharp/Microsoft.Spark/Sql/Functions.cs

```diff
@@ -4088,7 +4277,7 @@ private static UserDefinedFunction CreateVectorUdf<TResult>(string name, Delegat
                     CommandSerDe.SerializedMode.Row,
                     CommandSerDe.SerializedMode.Row),
                 evalType,
-                UdfUtils.GetReturnType(typeof(TResult)));
+                returnType?.Json ?? UdfUtils.GetReturnType(typeof(TResult)));
```

Comment:
Instead of mixing these two, we should have a separate function.

Reply:
Sure, I can create another function maybe called `CreateUdfWithReturnType`. Does the naming of this function make sense?